frontend/src/metabase/visualizations/components/legend/LegendItem.jsx

```diff
@@ -0,0 +1,89 @@
+import React, { memo, useCallback } from "react";
+import PropTypes from "prop-types";
+import {
+  LegendItemDot,
+  LegendItemLabel,
+  LegendItemRemoveIcon,
+  LegendItemRoot,
+  LegendItemTitle,
+} from "./LegendItem.styled";
+import Ellipsified from "metabase/components/Ellipsified";
+import Tooltip from "metabase/components/Tooltip";
+
+const propTypes = {
+  label: PropTypes.string,
+  index: PropTypes.number,
+  color: PropTypes.string,
+  isMuted: PropTypes.bool,
+  isNarrow: PropTypes.bool,
+  isVertical: PropTypes.bool,
+  onHoverChange: PropTypes.func,
+  onSelectSeries: PropTypes.func,
+  onRemoveSeries: PropTypes.func,
+};
+
+const LegendItem = ({
+  label,
+  index,
+  color,
+  isMuted,
+  isNarrow,
+  isVertical,
+  onHoverChange,
+  onSelectSeries,
+  onRemoveSeries,
+}) => {
+  const handleItemClick = useCallback(
+    event => {
+      onSelectSeries && onSelectSeries(event, index);
+    },
+    [index, onSelectSeries],
+  );
+
+  const handleItemMouseEnter = useCallback(
```

Comment:
It seems like these callbacks are being passed to `styled-components` components that are not memorized/pure. This makes `useCallback` redundant

Reply:
Fixed